### PR TITLE
Implementation to show details of undergrad universities for faculties

### PR DESCRIPTION
**Pull request by linhpha**
This PR adds functionality to view details of undergrad universities for faculties if they want to know more about the school students attended. The page also has Google Map API integrated to give the details on location. For Google Maps to work, we will need to have the API_KEY set up in the environment.

This PR contains commits from @hpitawela and @linhpha 

## Issue Related
#198 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Code coverage

96.15%

## Screenshot (if any change in front end)
image

image

### CODE REVIEW

File: app/assets/javascripts/undergrad_universities.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+var map;
+
+function initialize() {
+    // Create a map centered in Pyrmont, Sydney (Australia).
```
Comment (hpitawela) [inline app/assets/javascripts/undergrad_universities.js]: I have forgotten to change this comment. Can you make it:
// Create  a map centered in United States

File: app/assets/javascripts/undergrad_universities.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    map = new google.maps.Map(document.getElementById('map'), {
+        center: {lat: 36.2153013, lng: -113.6920816},
+        zoom: 5
+    });
+
+    // Search for Google's office in Australia.
```
Comment (hpitawela) [inline app/assets/javascripts/undergrad_universities.js]: And this one too:
// Search for university's location

Comment (linhpha) [reply app/assets/javascripts/undergrad_universities.js]: :+1:

File: app/assets/javascripts/undergrad_universities.js (hunk 1 above)
Comment (linhpha) [reply app/assets/javascripts/undergrad_universities.js]: :+1:

File: spec/controllers/undergrad_universities_controller_spec.rb [hunk 3]
```diff
@@ -69,5 +69,16 @@
       get :index
     end
   end
+
+  describe 'show method' do
+    before :each do
+      UndergradUniversity.create(id: 1, country_id: 2, university_name: 'University of Iowa')
+    end
+    it 'should find the university by id'do
+      undergrad_universities = UndergradUniversity.find_by_id(1)
```
Comment (Avanti0532) [inline spec/controllers/undergrad_universities_controller_spec.rb]: I think you won't need this as you are already mocking it in below line.

Comment (linhpha) [reply spec/controllers/undergrad_universities_controller_spec.rb]: :+1:

**Pull request merged**